src/core/reference/include/ngraph/runtime/reference/acosh.hpp

```diff
@@ -13,14 +13,14 @@ namespace reference {
 template <typename T, typename std::enable_if<!std::is_integral<T>::value, bool>::type = true>
 void acosh(const T* arg, T* out, size_t count) {
     for (size_t i = 0; i < count; i++) {
-        out[i] = std::acosh(arg[i]);
+        out[i] = static_cast<T>(std::acosh(arg[i]));
     }
 }
 
 template <typename T, typename std::enable_if<std::is_integral<T>::value, bool>::type = true>
 void acosh(const T* arg, T* out, size_t count) {
     for (size_t i = 0; i < count; i++) {
-        out[i] = std::roundl(std::acosh(arg[i]));
+        out[i] = static_cast<T>(std::roundl(std::acosh(arg[i])));
```

Comment:
should we use different functions depending on data type ? https://en.cppreference.com/w/cpp/numeric/math/acosh